sos/plugins/kernel.py

```diff
@@ -138,6 +139,9 @@ def setup(self):
             # and may also cause softlockups
             self.add_copy_spec("/proc/timer*")
 
+        if not self.get_option("trace"):
+            self.add_forbidden_path("/sys/kernel/debug/tracing/trace")
```

Comment:
How about just as "with-timer"? I think it's more consistent.

    if self.get_option("trace"):
        self.add_copy_spec("/sys/kernel/debug/tracing/trace")

Reply:
Personally I prefer it as a separate `kernel.trace`. The existing `with-timer` option is for different information and  it doesn't necessarily make sense to have both gated behind that option.

Reply:
I didn't mean to change the option from kernel.trace to kernel.with-timer. I meant to make the coding consistent with what is done for with-timer where add_copy_spec() is called only when necessary.

But there is an overlook to me that the pattern /sys/kernel/debug/tracing/* covers multiple files and it seems certainly verbose to write add_copy_spec() each file one by one without add_forbidden_path().

I then found add_forbidden_path() has already been used in kernel.py for tracing information:

        # FIXME: provide a a long-term solution for #1299
        self.add_forbidden_path([
            '/sys/kernel/debug/tracing/trace_pipe',
            '/sys/kernel/debug/tracing/README',
            '/sys/kernel/debug/tracing/trace_stat/*',
            '/sys/kernel/debug/tracing/per_cpu/*',
            '/sys/kernel/debug/tracing/events/*',
            '/sys/kernel/debug/tracing/free_buffer',
            '/sys/kernel/debug/tracing/trace_marker',
            '/sys/kernel/debug/tracing/trace_marker_raw',
            '/sys/kernel/debug/tracing/instances/*/per_cpu/*/snapshot_raw',
            '/sys/kernel/debug/tracing/instances/*/per_cpu/*/trace_pipe*',
            '/sys/kernel/debug/tracing/instances/*/trace_pipe'
        ])

so I think Mizuta-san's first patch is consistent enough this way.